pkg/resources/task_acceptance_test.go

```diff
@@ -1,564 +1,970 @@
+						HasSchedule("10 MINUTES").
+						HasPredecessors().
+						HasState(sdk.TaskStateStarted).
+						HasDefinition(statement).
+						HasCondition(condition).
+						HasAllowOverlappingExecution(true).
+						HasErrorIntegration(errorNotificationIntegration.ID().Name()).
+						HasLastCommittedOnNotEmpty().
+						HasLastSuspendedOn("").
+						HasOwnerRoleType("ROLE").
+						HasConfig(expectedTaskConfig).
+						HasBudget("").
+						HasTaskRelations(sdk.TaskRelations{}),
+					resourceparametersassert.TaskResourceParameters(t, configModel.ResourceReference()).
+						HasAllDefaults(),
+				),
+			},
+			{
+				ResourceName: "snowflake_task.test",
+				ImportState:  true,
+				ImportStateCheck: assert.AssertThatImport(t,
+					resourceassert.ImportedTaskResource(t, helpers.EncodeResourceIdentifier(id)).
+						HasFullyQualifiedNameString(id.FullyQualifiedName()).
+						HasDatabaseString(id.DatabaseName()).
+						HasSchemaString(id.SchemaName()).
+						HasNameString(id.Name()).
+						HasEnabledString(r.BooleanTrue).
+						HasWarehouseString(acc.TestClient().Ids.WarehouseId().Name()).
+						HasScheduleString("10 MINUTES").
+						HasConfigString(expectedTaskConfig).
+						HasAllowOverlappingExecutionString(r.BooleanTrue).
+						HasErrorIntegrationString(errorNotificationIntegration.ID().Name()).
+						HasCommentString(comment).
+						HasFinalizeString("").
+						HasNoAfter().
+						HasWhenString(condition).
+						HasSqlStatementString(statement),
+				),
 			},
 		},
+	})
+}
 
-		ChildTask: &TaskSettings{
-			Name:    childname,
-			SQL:     "SELECT 1",
-			Enabled: true,
-			Comment: "third state",
-		},
+func TestAcc_Task_Updates(t *testing.T) {
+	_ = testenvs.GetOrSkipTest(t, testenvs.EnableAcceptance)
+	acc.TestAccPreCheck(t)
+
+	currentRole := acc.TestClient().Context.CurrentRole(t)
+
+	id := acc.TestClient().Ids.RandomSchemaObjectIdentifier()
+	statement := "SELECT 1"
+	basicConfigModel := model.TaskWithId("test", id, false, statement)
+
+	// New warehouse created, because the common one has lower-case letters that won't work
+	warehouse, warehouseCleanup := acc.TestClient().Warehouse.CreateWarehouse(t)
+	t.Cleanup(warehouseCleanup)
+
+	errorNotificationIntegration, errorNotificationIntegrationCleanup := acc.TestClient().NotificationIntegration.Create(t)
+	t.Cleanup(errorNotificationIntegrationCleanup)
 
-		SoloTask: &TaskSettings{
-			Name:              soloname,
-			Schema:            acc.TestSchemaName,
-			SQL:               "SELECT *",
-			When:              "FALSE",
-			Enabled:           true,
-			Schedule:          "15 MINUTE",
-			UserTaskTimeoutMs: 900000,
+	taskConfig := `$${"output_dir": "/temp/test_directory/", "learning_rate": 0.1}$$`
+	// We have to do three $ at the beginning because Terraform will remove one $.
+	// It's because `${` is a special pattern, and it's escaped by `$${`.
+	expectedTaskConfig := strings.ReplaceAll(taskConfig, "$", "")
```

Comment:
if there is some repeated escaping logic for tasks configs please extract it either as a function here or in some test helper

Reply:
I added this point to `next prs`. I would like to remove the need to put `$$` in the config field. It's repetitive and the user should be able to put any kind of JSON straight into this field without this strange escaping (especially when it's colliding with HCL syntax).

Reply:
oky